WordPress/Classes/Services/MediaImportService.swift

```diff
@@ -20,55 +20,254 @@ open class MediaImportService: LocalCoreDataService {
     ///
     @objc static let preferredImageCompressionQuality = 0.9
 
-    /// Allows the caller to designate supported import file types
-    @objc var allowableFileExtensions = Set<String>()
-
     static let defaultAllowableFileExtensions = Set<String>(["docx", "ppt", "mp4", "ppsx", "3g2", "mpg", "ogv", "pptx", "xlsx", "jpeg", "xls", "mov", "key", "3gp", "png", "avi", "doc", "pdf", "gif", "odt", "pps", "m4v", "wmv", "jpg"])
 
     /// Completion handler for a created Media object.
     ///
-    public typealias MediaCompletion = (Media) -> Void
+    typealias MediaCompletion = (Media) -> Void
 
     /// Error handler.
     ///
-    public typealias OnError = (Error) -> Void
+    typealias OnError = (Error) -> Void
+
+    private let coreDataStack: CoreDataStackSwift
+
+    /// The initialiser for Objective-C code.
+    ///
+    /// Using `ContextManager` as the argument becuase `CoreDataStackSwift` is not accessible from Objective-C code.
+    @objc
+    convenience init(contextManager: ContextManager) {
+        self.init(coreDataStack: contextManager)
+    }
+
+    init(coreDataStack: CoreDataStackSwift) {
+        self.coreDataStack = coreDataStack
+    }
 
     // MARK: - Instance methods
 
+    /// Create a media object using the url provided as the source of media.
+    ///
+    /// - Note: All blocks arguments are called from the main thread. The `Media` argument in the blocks is bound to
+    ///     the main context.
+    ///
+    /// - Warning: This function must be called from the main thread.
+    ///
+    /// This functions returns a `Media` instance. To ensure the returned `Media` instance continue to be a valid
+    /// instance, it can't be bound to a background context which are all temporary context. The only long living
+    /// context is the main context. And the safe way to create and return an object bound to the main context is
+    /// doing it from the main thread, which is why this function must be called from the main thread.
+    ///
+    /// - Parameters:
+    ///   - exportable: an object that implements the exportable interface
+    ///   - blog: the blog object to associate to the media
+    ///   - post: the post object to associate to the media
```

Comment:
```suggestion
    ///   - post: the optional post object to associate to the media
```

I'm guessing the post is optional because we can upload assets via the media library itself, in which case they wouldn't be used in any post?